File: src/lib/models/DTO/Domain/MeetingSummaryQueryBuilderConfigDto.cs
namespace DTO.Domain
{
    public class MeetingSummaryQueryBuilderConfigDto
    {
        public short MeetingStatusId{ get; set; }
        public System.Guid? CreationUserUniqueId{ get; set; }

        public string GenerateCacheKey() 
            => $"{nameof(MeetingSummaryQueryBuilderConfigDto)}::" +
            $"{nameof(this.MeetingStatusId)}:{this.MeetingStatusId};" + 
            $"{nameof(this.CreationUserUniqueId)}:{this.CreationUserUniqueId}";
    }
}

File: src/ui/asq-ui/src/app/_models/domain/meeting-summary-query-builder-config-dto.ts
export class MeetingSummaryQueryBuilderConfigDto{
    public meetingStatusId: number = 1;
    public creationUserUniqueId: string = null;

    constructor(){ }

    public generateCacheKey = () => `MeetingSummaryQueryBuilderConfig::MeetingStatusId:${this.meetingStatusId};CreationUserUniqueId:${this.creationUserUniqueId}`;
}
